Kiosk/App/SwiftExtensions.swift

```diff
@@ -17,8 +17,40 @@ public extension String {
     public func toUIntWithDefault(defaultValue: UInt) -> UInt {
         return UInt(self) ?? defaultValue
     }
+}
+
+// Anything that can hold a value (strings, arrays, etc)
+protocol Occupiable {
+    var isEmpty: Bool { get }
+    var isNotEmpty: Bool { get }
+}
 
-    public var isNotEmpty: Bool {
+// Give a default implementation of isNotEmpty, so conformance only requires one implementation
+extension Occupiable {
+    var isNotEmpty: Bool {
         return !isEmpty
     }
 }
+
+extension String: Occupiable { }
+
+// I can't think of a way to combine these collection types. Suggestions welcome.
+extension Array: Occupiable { }
+extension Dictionary: Occupiable { }
+extension Set: Occupiable { }
```

Comment:
It seems extending protocols to conform to other protocols isn't implemented yet. https://twitter.com/jckarter/status/616254698509611008